src/coordinator/functions/abs.go

```diff
@@ -59,25 +59,65 @@ type AbsNode struct {
 
 // Process the block
 func (c *AbsNode) Process(ID parser.NodeID, b block.Block) error {
-	builder, err := c.controller.BlockBuilder(b.Meta(), b.SeriesMeta())
+	stepIter, err := b.StepIter()
 	if err != nil {
 		return err
 	}
 
-	stepIter := b.StepIter()
-	if err := builder.AddCols(b.StepCount()); err != nil {
+	builder, err := c.controller.BlockBuilder(stepIter.Meta(), stepIter.SeriesMeta())
+	if err != nil {
+		return err
+	}
+
+	if err := builder.AddCols(stepIter.StepCount()); err != nil {
 		return err
 	}
 
 	for index := 0; stepIter.Next(); index++ {
-		step := stepIter.Current()
-		values := step.Values()
+		step, err := stepIter.Current()
+		if err != nil {
+			return err
+		}
+
+		values := c.process(step.Values())
 		for _, value := range values {
-			builder.AppendValue(index, math.Abs(value))
+			builder.AppendValue(index, value)
 		}
 	}
 
 	nextBlock := builder.Build()
 	defer nextBlock.Close()
 	return c.controller.Process(nextBlock)
 }
+
+// Ensure AbsNode implements the types
+var _ transform.StepNode = &AbsNode{}
+var _ transform.SeriesNode = &AbsNode{}
+
+// ProcessStep allows step iteration
+func (c *AbsNode) ProcessStep(step block.Step) (block.Step, error) {
+	return block.NewColStep(step.Time(), c.process(step.Values())), nil
```

Comment:
can we pull out `processedValues := c.process(step.Values())`, to make it very clear what we're doing here?

Reply:
+1